eradiate/radprops/mie.py

```diff
@@ -0,0 +1,318 @@
+"""
+Mie scattering properties computation.
+"""
+import typing as t
+
+import miepython
+import numpy as np
+import pint
+import xarray as xr
+
+
+from ..units import unit_registry as ureg
+
+_DEFAULT_MU = np.linspace(-1, 1, 201)
+
+
+@ureg.wraps(ret=None, args=("nm", None, None, "dimensionless"), strict=False)
+def compute_properties(
+    w: t.Union[ureg.Quantity, np.ndarray],
+    rdist: xr.DataArray,
+    m: xr.DataArray,
+    mu: t.Union[ureg.Quantity, np.ndarray] = _DEFAULT_MU,
```

Comment:
You should use `np.typing.ArrayLike` for params accepting anything that can be converted to a Numpy array. Be sure to double check in that makes sense; you'd then have:
```suggestion
    w: t.Union[ureg.Quantity, np.typing.ArrayLike],
    rdist: xr.DataArray,
    m: xr.DataArray,
    mu: t.Union[ureg.Quantity, np.typing.ArrayLike] = _DEFAULT_MU,
```